fuel/utils/formats.py

```diff
@@ -1,8 +1,35 @@
 """Low-level utilities for reading a variety of source formats."""
+import codecs
+import gzip
 import tarfile
 import six
 
 
+def open_(filename, mode='r', encoding='utf-8'):
+    """Open a text file with UTF-8 and optional gzip compression.
+
+    This function is useful when dealing with text files.
+
+    Parameters
+    ----------
+    filename : str
+        The filename to read.
+    mode : str, optional
+        The mode with which to open the file. Defaults to `r`.
+    encoding : str, optional
+        The encoding to use (see the codecs documentation_ for supported
+        values). Defaults to `utf-8`.
+
+    .. _documentation:
+    https://docs.python.org/3/library/codecs.html#standard-encodings
+
+    """
+    if filename.endswith('.gz.'):
+        zf = gzip.open(filename, mode)
```

Comment:
I already tried this gzip and it was very slow to compute. It use a too small buffer to uncompress. It was way faster to uncompress on the command line wigh gunzip. Did you see speed up with this compared to regular file? If so, with which version of Python?